### PR TITLE
Accountant

### PR DESCRIPTION
**Pull request by BobTheBuidler**

### CODE REVIEW

File: yearn/treasury/accountant/cost_of_revenue/general.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            if any(tx.token.address.address == convert.to_address(wrapper.vault) for wrapper in partner.flat_wrappers):
+                return True
+    
+    # DEV figure out why these weren't captured by the above
+    hashes = {
+        Network.Mainnet: [
+            # Thought we automated these... why aren't they sorting successfully? 
+            ["0x590b0cc67ba42dbc046b8cbfe2d314fbe8da82f11649ef21cdacc61bc9752d83", IterFilter('log_index',[275,276,278])],
+            ["0xd1b925ad7fdd9abdd31460a346d081d6afe9f6cb1c1b0cd5f6129885edf318da", IterFilter('log_index',[174,177])],
+            ["0xe11b4e3ece520c1818ffe821c038779f87c293aa32c26115265b6b8fb23c30bd", Filter('log_index', 154)],
```
Comment (mariuspod) [inline yearn/treasury/accountant/cost_of_revenue/general.py:49]: what are these magic numbers here ?

**Review by mariuspod — CHANGES_REQUESTED** [review]
I've added some comments to some of your changes mainly targeting custom workarounds or magic numbers which are not critical.

The only critical thing is the exposure of the db port to the world inside docker-compose which I'd rather not do.

**Comment by BobTheBuidler** [comment]
> I've added some comments to some of your changes mainly targeting custom workarounds or magic numbers which are not critical.
> 
> The only critical thing is the exposure of the db port to the world inside docker-compose which I'd rather not do.

ah yes, that was just supposed to be a local change, will revert

File: yearn/treasury/accountant/cost_of_revenue/general.py (hunk 1 above)
Comment (BobTheBuidler) [reply yearn/treasury/accountant/cost_of_revenue/general.py:49]: They are the log indexes of the particular value change components of each tx that need to be sorted

File: yearn/__init__.py [hunk 2]
```diff
@@ -4,6 +4,11 @@
 
 setup_logging()
 setup_sentry()
-if network.is_connected():
+# hacky workaround when node doesn't support web3_clientVersion
+try:
+    if network.is_connected():
+        from yearn.middleware.middleware import setup_middleware
+        setup_middleware()
+except AssertionError:
```
[the remark this answers is not in the document]
Comment (BobTheBuidler) [reply yearn/__init__.py:12]: Our nodes, or nodes generally? 
I don't recall offhand which one, but this was a fix for a node from one of the saas providers

File: services/dashboard/docker-compose.yml [hunk 3]
```diff
@@ -629,7 +700,7 @@ services:
   postgres:
     image: postgres
     ports:
-      - 127.0.0.1:5432:5432
+      - 5432:5432
```
[the remark this answers is not in the document]
Comment (BobTheBuidler) [reply services/dashboard/docker-compose.yml:703]: yep that was an accident, will revert

File: scripts/treasury_transactions_exporter.py [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+
+warnings.simplefilter("ignore", BrownieEnvironmentWarning)
+
+logger = logging.getLogger('yearn.treasury_transactions_exporter')
+
+BATCH_SIZE = {
+    Network.Mainnet: 25_000,
+    Network.Fantom: 2_000_000,
+    Network.Gnosis: 2_000_000,
+}[chain.id]
```
[the remark this answers is not in the document]
Comment (BobTheBuidler) [reply scripts/treasury_transactions_exporter.py:41]: Ah yes, arb was added after this PR was started. Will add.

**Comment by BobTheBuidler** [comment]
> > I've added some comments to some of your changes mainly targeting custom workarounds or magic numbers which are not critical.
> > The only critical thing is the exposure of the db port to the world inside docker-compose which I'd rather not do.
> 
> ah yes, that was just supposed to be a local change, will revert

reverted and pushed, marking this as resolved.

File: services/dashboard/docker-compose.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (BobTheBuidler) [reply services/dashboard/docker-compose.yml]: reverted, resolving

File: Makefile [hunk 5]
```diff
@@ -3,11 +3,46 @@ ifdef FLAGS
...
+
+treasury_containers := $(ethereum_treasury_containers) $(fantom_treasury_containers) $(arbitrum_treasury_containers) $(gnosis_treasury_containers)
+all_containers := $(ethereum_containers) $(fantom_containers) $(arbitrum_containers) $(gnosis_containers)
+
+# Basic Makefile commands
+dashboards: dashboards-up
+tvl: tvl-up
+
+logs:
+	$(dashboards_command) logs -f -t $(ethereum_containers_lite) $(fantom_containers_lite) $(arbitrum_containers_lite) $(gnosis_containers_lite)
```
[the remark this answers is not in the document]
Comment (BobTheBuidler) [reply Makefile:37]: Resolving per discussions with @mariuspod  in DM's

File: yearn/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mariuspod) [reply yearn/__init__.py:12]: I checked the code again, `web3.clientVersion` is referenced in here in the sentry init code: https://github.com/yearn/yearn-exporter/blob/3a4ab9873dee063cb38e30cb387e0af0ad2343ca/yearn/sentry.py#L18

So maybe the `try..except` must be expanded as `setup_sentry` is called before it ?
You can also try to set the the clientVersion to some value if it's not available or create a check that it's not sent to sentry if n/a.

**Pull request merged**